sdk/cosmos/azure-cosmos/azure/cosmos/_query_iterable.py

```diff
@@ -90,14 +97,45 @@ def PartitioningQueryIterable(cls, client, query, options, database_link, partit
     def _create_execution_context(self):
         """instantiates the internal query execution context based.
         """
-        if hasattr(self, "_database_link"):
-            # client side partitioning query
-            return base_execution_context._MultiCollectionQueryExecutionContext(
-                self._client, self._options, self._database_link, self._query, self._partition_key
-            )
-        return execution_dispatcher._ProxyQueryExecutionContext(
-            self._client, self._collection_link, self._query, self._options, self._fetch_function
-        )
+        query_execution_context = _DefaultQueryExecutionContext(self._client, self._options, self._fetch_function)
+        if self._resource_type != http_constants.ResourceType.Document or self._use_default_query_execution_context():
```

Comment:
We have to use default for the first execution context because of the scenario:

Orderby with partition key in the query string itself (not in options). Because we allow this, we cannot determine if the gateway will serve a query or not until we send it. If it throws, then we switch to query plan.